Update module timeout env var instructions to set vars locally (#4234)
viamrobotics/docs: docs/dev/tools/common-errors.md

**Solution:**

- Try using a faster internet connection.
- If you are the module author, consider packaging the module with required dependencies so they don't need to be downloaded on startup.
  For Python modules, you can package your module with dependencies by using the PyInstaller steps when [uploading your module](/operate/get-started/other-hardware/#upload-your-module).

> - If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine using the `viam-server` service configuration file.

Author: ```suggestion
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine to a higher value. You can set these environment variables when you start `viam-server`, for instance `VIAM_MODULE_STARTUP_TIMEOUT=6m30 VIAM_RESOURCE_CONFIGURATION_TIMEOUT=3m0s viam-server -config example-machine.json`.
```

>   Pass a sequence of numbers and time units, for example "6m30s50ms" for a timeout of 6 minutes, 30 seconds, and 50 milliseconds, or "5m" for a timeout of 5 minutes.

Reviewer: Ooh by a full example I meant like `VIAM_MODULE_STARTUP_TIMEOUT=6m30 viam-server -config xxx.json` or `Environment="VIAM_MODULE_STARTUP_TIMEOUT=6m30s"` (for each of the two  current ways to do this). Do you think we shouldn't show the command line way?

Author: yep, i think we're on the same wavelength -- while I would _like_ to show the command line way, it requires a bit too much explanation so I think we're safer distancing ourselves from it a bit. Especially since this is a temporary situation and we'll hopefully have the JSON configuration back as best practice in a bit.

Reviewer: Seems like from what Cheuk said if we are only going to show one way it should be the command line way and not the service file way, but we could show both

Reviewer: But also yes temporary so I don't suppose it matters!

Author: Thanks for this feedback, I added a CLI example showing how to configure this during manual `viam-server` startup. Assuming our checks pass, I'll merge this in so we don't have any more outdated info in the docs. This solution isn't perfect, but we know we'll update this eventually to demonstrate a better solution anyway. Good enough for this interim period.

**Solution:**

- Try using a faster internet connection.
- If you are the module author, consider packaging the module with required dependencies so they don't need to be downloaded on startup.
  For Python modules, you can package your module with dependencies by using the PyInstaller steps when [uploading your module](/operate/get-started/other-hardware/#upload-your-module).
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine using the `viam-server` service configuration file.

>   To load new environment variable values, restart `viam-server` from the Viam app.
>   For a value, pass a sequence of numbers and units, for example "6m30s50ms" or "5m".

Author: ```suggestion
  Pass a sequence of numbers and time units, for example "6m30s50ms" for a timeout of 6 minutes, 30 seconds, and 50 milliseconds, or "5m" for a timeout of 5 minutes.
  To load new environment variable values, restart `viam-server` from the Viam app.
```
Swapping the order of these sentences and including a translation from machine time unit specification to human time units.

> 

Author: ```suggestion
```
trim whitespace line to appease linter

>   To load new environment variable values, restart `viam-server` from the Viam app.

Reviewer: Wait but then this sentence doesn't make sense anymore *insert spiral emoji and lolsob*

**Solution:**

- Try using a faster internet connection.
- If you are the module author, consider packaging the module with required dependencies so they don't need to be downloaded on startup.
  For Python modules, you can package your module with dependencies by using the PyInstaller steps when [uploading your module](/operate/get-started/other-hardware/#upload-your-module).

> - If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine.

Reviewer: I worry that if we say this without saying *how* to set the env vars, and then continue to link to the module config details page below which shows using them with the config, people may make the same assumption I did that they can be used within the config.
We could clarify here but maybe overkill depending on how long it takes for the eng fix to go in.
```suggestion
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine.
  The environment variables will not be picked up from the JSON configuration until after the module has already started up and configured, so you must set the variables either from your command line or in the `viam-server` service config file.
```

Author: Great point. Unfortunately this still implies that the JSON configuration works, but unfortunately it doesn't sound like that is the case right now.
What do you think of also deleting the sentence below that refers to the module config details page? That way, users will only see a reference to environment variables, and they won't be biased into thinking that they should use the (nonfunctional) JSON config to set them.

We can bring the link-containing sentence back once engineering makes the JSON config for env vars functional.

Author: ```suggestion
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine using the `viam-server` service configuration file.
  To load new environment variable values, restart `viam-server` from the Viam app.
```
To avoid mentioning the tricksy JSON, and make sure users actually load the vars, I trimmed your language down and added an extra instruction.

Reviewer: This suggestion seems to disregard Cheuk's suggested command line method though?

Author: I couldn't get the command line method to work on any of my deployments, so I'm not comfortable recommending it to users 🥵 . If you know a command line method that works, I'm happy to incorporate it, but Cheuk and I both felt like the bash CLI approach was a little too error-prone!

>   For a value, pass sequence of numbers and a units, e.g. "6m30s", "6.5m", or "630000ms".

Author: ```suggestion
  For a value, pass a sequence of numbers and units, e.g. "6m30s", "6.5m", or "630000ms".
```

>   For a value, pass a sequence of numbers and units, e.g. "6m30s", "6.5m", or "630000ms".

Reviewer: ```suggestion
  For a value, pass a sequence of numbers and units, for example `6m30s`, `6.5m`, or `630000ms`.
```
(Or optionally, include quotes inside the backticks if the thing must be passed with quotes as a string)

Author: ```suggestion
  For a value, pass a sequence of numbers and units, for example "6m30s50ms" or "5m".
```

> For a value, pass sequence of numbers and a units, e.g. "6m30s", "6.5m", or "630000ms".
> You can specify units in "ns" (nanoseconds), "us"/"µs" (microseconds), "ms" (milliseconds), "s" (seconds), "m" (minutes), and "h" (hours). 

Author: ```suggestion
  For a value, pass sequence of numbers and a units, e.g. "6m30s", "6.5m", or "630000ms".
  You can specify units in "ns" (nanoseconds), "us"/"µs" (microseconds), "ms" (milliseconds), "s" (seconds), "m" (minutes), and "h" (hours). 
```

>   You can specify units in "ns" (nanoseconds), "us"/"µs" (microseconds), "ms" (milliseconds), "s" (seconds), "m" (minutes), and "h" (hours). 

Reviewer: Might actually be more useful with fewer words to show an example of the complete syntax you'd pass in the command line and/or in the service config file, and not worry about the different possible time units since those seem like general convention?

Author: Great point. I'll just show one example with m, s, and ms to keep things brief. lord help us if someone sets a timeout in ns 😆

>   You can specify units in "ns" (nanoseconds), "us"/"µs" (microseconds), "ms" (milliseconds), "s" (seconds), "m" (minutes), and "h" (hours). 
>   For more information on environment variables used by modules, see [Module Configuration Details](/operate/reference/module-configuration/#environment-variables).

Author: ```suggestion
```
Let's remove this entirely until we fix the JSON configuration.

**Solution:**

- Try using a faster internet connection.
- If you are the module author, consider packaging the module with required dependencies so they don't need to be downloaded on startup.
  For Python modules, you can package your module with dependencies by using the PyInstaller steps when [uploading your module](/operate/get-started/other-hardware/#upload-your-module).
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine.

>   To configure these variables on a Linux machine that uses the `bash` shell:
> 
>   1. Connect to your machine using SSH.
>   1. Open your user account's <file>~/.bashrc</file> file.
>   1. Add the following lines to the end of the file:
> 
>      ```bash
>      export VIAM_MODULE_STARTUP_TIMEOUT="6m30s"
>      export VIAM_RESOURCE_CONFIGURATION_TIMEOUT="3m0s"
>      ```
> 
>   1. Run `source ~/.bashrc` to load your `bash` configuration changes.
>   1. Finally, run `sudo systemctl restart viam-server` to restart `viam-server`.
>  

Author: ```suggestion
For a value, pass sequence of numbers and a units, e.g. "6m30s", "6.5m", or "630000ms".
You can specify units in "ns" (nanoseconds), "us"/"µs" (microseconds), "ms" (milliseconds), "s" (seconds), "m" (minutes), and "h" (hours). 
```
Implementing a suggestion: removing the explicit instructions entirely. Providing a little more information about the units that viam-server can parse for env var time duration input. ns & us & h are probably overkill, but they don't hurt.

- Try using a faster internet connection.
- If you are the module author, consider packaging the module with required dependencies so they don't need to be downloaded on startup.
  For Python modules, you can package your module with dependencies by using the PyInstaller steps when [uploading your module](/operate/get-started/other-hardware/#upload-your-module).
- If the problem persists, try setting the `VIAM_MODULE_STARTUP_TIMEOUT` or `VIAM_RESOURCE_CONFIGURATION_TIMEOUT` environment variables on your machine.
  To configure these variables on a Linux machine that uses the `bash` shell:

  1. Connect to your machine using SSH.
  1. Open your user account's <file>~/.bashrc</file> file.
  1. Add the following lines to the end of the file:

     ```bash
     export VIAM_MODULE_STARTUP_TIMEOUT="6m30s"
     export VIAM_RESOURCE_CONFIGURATION_TIMEOUT="3m0s"
     ```

  1. Run `source ~/.bashrc` to load your `bash` configuration changes.
  1. Finally, run `sudo systemctl restart viam-server` to restart `viam-server`.
 
  For more information on environment variables used by modules, see [Module Configuration Details](/operate/reference/module-configuration/#environment-variables).

> 

Author: ```suggestion
```
remove extra line, lint does not like
